### ISSUE TITLE
Machine Learning

### ISSUE THREAD

**Issue by innerlee** [innerlee/Digest#4]
- Statistical learning theory shows that the discrepancy between training error and generalization error is bounded from above by a quantity that grows as the model capacity grows but shrinks as the number of training examples increases.
- _No free lunch theorem_. Averaged over all possible data generating distributions, every classification algorithm has the same error rate when classifying previously unobserved points. In other words, no machine learning algorithm is universally any better than any other. So we care about machine learning algorithms that perform well on data drawn from distributions we care about.
- _Weight decay_ expresses a preference for the weights to have smaller squared `L^2` norm.
- _Regularization_ is any modification we make to a learning algorithm that is intended to reduce its generalization error but not its training error.
- _Validation set_ is used to train the hyper-parameters. Typically, one uses about 80% of the training data for training and 20% for validation.
- _Maximum likelihood estimation_ can be viewed as minimizing the dissimilarity (KL divergence) between the empirical distribution defined b the training set and the model distribution.
- Minimizing this KL divergence corresponds to minimizing the cross entropy between the distributions.
- Any loss consisting of a negative log-likelihood is a cross entropy between the empirical distribution and the model.
- Mean squared error is the cross entropy between the empirical distribution and a Gaussian model.
- _MAP_ Bayesian inference with a Gaussian prior on the weights corresponds to weight decay.
- The core idea in deep learning is that we assume that the data was generated by the _composition of factors_ or features, potentially at multiple levels in a hierarchy.
- The probability distribution over images, text strings, and sounds that occur in real life is highly concentrated.
